server/src/main/java/io/crate/planner/operators/HashJoinConditionSymbolsExtractor.java

```diff
@@ -92,7 +92,7 @@ public Void visitFunction(Function function, Context context) {
                     int duplicatePos = 0;
                     for (Symbol arg : function.arguments()) {
                         var relation = arg.accept(RELATION_EXTRACTOR, null);
-                        List<Symbol> symbols = context.symbolsPerRelation.computeIfAbsent(relation, k -> new ArrayList<>());
+                        List<Symbol> symbols = context.symbolsPerRelation.computeIfAbsent(relation, k -> new LinkedList<>());
```

Comment:
Is this change necessary? I don't understand how this helps, `ArrayList` also keeps the insertion order

Reply:
it slipped in, it is not needed.